design/user-id-update.md

```diff
@@ -0,0 +1,44 @@
+## Make User IDs persistent across session ID, but unique per AppID
+
+This is related to github issue #129.
+
+I want us to redo user IDs and persistence. Current implementation is inconsistent across testbed+Altspace, is inconsistent across registered/unregistered apps, and was designed this way based off of us not having a permission system.
+
+For Background for current implementation, see Issue #129:
+1. Eric wrote: "In a space where an MRE is instantiated multiple times with the same sessionId, a given user in the world will have a different userId in each instance."
+2. Steven wrote this comment : "@eanders-MS This is by design. GDPR requires us to not allow users to be tracked without their permission. As such, any given Altspace user will be assigned a different ID for each instance of a given app. I agree that a persistent ID per app would be tremendously useful, but we have to come up with a strategy to protect our users' privacy first."
+3. Eric wrote about a design:
+	"Registered apps: Same userId for a given user across multiple instances of an app (because the AppId is consistent).
+	Unregistered apps: Different userIds for a given user across multiple instances of an app (because the AppId is derived from the space component id)."
+	"We talked about adding functionality on the website for users to be able to be forgotten by all apps. How this would work: When generating the userId, include a key taken from the user record. When the user chooses to be forgotten, we would roll this key"
+
+Currently in Altspace we use: UserID= combination of altspaceUserID, sessionID, and SCID
+
+Currently in testbed we use: UserID=Random generated every connection
+
+### Proposed design
+Just have user runtime hash userID with an AppID, not with sessionID/SCID/anything else. And have the hashing logic be part of the unity runtime instead of be the host app's responsibility
+
+(Note about AppID: AppID for now is URL, but in the future AppID should be either URL or actual AppID registered in the hostapp-agnostic MRE registry - see issue #617)
```

Comment:
AppID is a long int assigned to the registration by the Altspace backend, not a URL. Unless you're referring to the new registry ID here?

Reply:
Correct, this was unclear because I was thinking "new registry". Clarified, and made sure we can support both current altspace registry for now and eventual new registry.